src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -589,7 +589,9 @@ DeferErrorIfUnsupportedSubqueryPushdown(Query *originalQuery,
 								 NULL);
 		}
 	}
-	else if (!RestrictionEquivalenceForPartitionKeys(plannerRestrictionContext))
+	else if (!RestrictionEquivalenceForPartitionKeys(plannerRestrictionContext) ||
```

Comment:
Can you explain the reason why we ended-up adding `AllDistributedRelationsInRestrictionContextColocated` all over the planner, and we expanded equality checks with colocation checks?

I really doubt we need it in so many places.  Can you, for example, share the example queries that are fixed by this?

ps: we already have colocation checks in place, otherwise we'd return wrong results.

Reply:
As we've discussed, this change has been reverted.